### PR TITLE
Support wagtail42

### PR DESCRIPTION
**Pull request by nickmoreton**
This updates the package to support Wagtail v4.2

Drops support for wagtail < 4.1

The following PR's have been included here.

- https://github.com/wagtail/wagtail-personalisation/pull/244

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,66 +1,76 @@
...
 from setuptools import find_packages, setup
 
 install_requires = [
-    'wagtail>=2.0',
-    'user-agents>=1.1.0',
-    'wagtailfontawesome>=1.1.3',
-    'pycountry',
+    "wagtail>=4.1",
+    "user-agents>=1.1.0",
+    "wagtail-font-awesome-svg>=0.0.3",
```
Comment (victoriachan) [inline setup.py]: When testing this locally, I got the error `ModuleNotFoundError: No module named 'wagtailfontawesome'` when running `./manage.py migrate`. The old migration files still refer to wagtailfontawesome.

**Review by victoriachan — CHANGES_REQUESTED** [review]
Looks good but I got an error about wagtailfontawesome

File: setup.py (hunk 1 above)
Comment (nickmoreton) [reply setup.py]: I'm not sure that now `wagtail-font-awesome-svg>=0.0.3` should have been changed. I think I assumed this package replaced the previous one. But they are different packages.

Did you see the error when running the sandbox or by adding it to a new site?

Comment (nickmoreton) [reply setup.py]: @victoriachan I have reverted the change that switched from `wagtailfontawesome` to `wagtail-font-awesome-svg` as that looks like a mistake.

I also needed to update `django-debug-toolbar`

These changes seem to work for me, can you try again please?

Comment (victoriachan) [reply setup.py]: Ah that works now. Thanks. I was previously testing on a wagtail kit build. I see that you have fixed the sandbox and it works fine for me now too. 

I have got some warnings about `DEFAULT_AUTO_FIELD` on my console (for the kit site) though. E.g.

```
 wagtail_personalisation.DayRule: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
 	HINT: Configure the DEFAULT_AUTO_FIELD setting or the AppConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
 wagtail_personalisation.DeviceRule: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
 	HINT: Configure the DEFAULT_AUTO_FIELD setting or the AppConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
 wagtail_personalisation.OriginCountryRule: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
 	HINT: Configure the DEFAULT_AUTO_FIELD setting or the AppConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
 wagtail_personalisation.PersonalisablePageMetadata: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
 	HINT: Configure the DEFAULT_AUTO_FIELD setting or the AppConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
 wagtail_personalisation.QueryRule: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
 	HINT: Configure the DEFAULT_AUTO_FIELD setting or the AppConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
 wagtail_personalisation.ReferralRule: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
 	HINT: Configure the DEFAULT_AUTO_FIELD setting or the AppConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
 wagtail_personalisation.Segment: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
 	HINT: Configure the DEFAULT_AUTO_FIELD setting or the AppConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
 wagtail_personalisation.TimeRule: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
 	HINT: Configure the DEFAULT_AUTO_FIELD setting or the AppConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
 wagtail_personalisation.UserIsLoggedInRule: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
 	HINT: Configure the DEFAULT_AUTO_FIELD setting or the AppConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
 wagtail_personalisation.VisitCountRule: (models.W042) Auto-created primary key used when not defining a primary key type, by default 'django.db.models.AutoField'.
 	HINT: Configure the DEFAULT_AUTO_FIELD setting or the AppConfig.default_auto_field attribute to point to a subclass of AutoField, e.g. 'django.db.models.BigAutoField'.
```

I guess you need to add default_auto_field to your app config?

**Comment by victoriachan** [comment]
New changes look good, just need to fix warning about default_auto_field. Also the Readme needs updating as it says "Wagtail Personalisation requires Wagtail 2.0 or 2.1 and Django 1.11 or 2.0."!

File: src/wagtail_personalisation/__init__.py [hunk 2]
```diff
@@ -1 +1 @@
-# default_app_config = 'wagtail_personalisation.config.WagtailPersonalisationConfig'
+default_app_config = 'wagtail_personalisation.config.WagtailPersonalisationConfig'
```
Comment (nickmoreton) [inline src/wagtail_personalisation/__init__.py]: @victoriachan I must have left this commented out during testing. Can you see if it fixes the issue as I wasn't seeing the console message. Thanks

**Review by nickmoreton — COMMENTED** [review]
@victoriachan I have fixed the console waring hopefully. Also updated the docs.
The imagery used in the docs is not right in terms of the look of the admin. Do you think we should update them?

**Comment by katdom13** [comment]
Hi @nickmoreton ,
I'm not seeing the warning. Maybe these comments can be resolved?

**Review by katdom13 — APPROVED** [review]

**Review by katdom13 — APPROVED** [review]

**Comment by easherma-truth** [comment]
Any plans to merge this or the 5.2 PR soon?

**Comment by zerolab** [comment]
Closing in favour of #251

**Pull request closed without merging**